src/virt.c

```diff
@@ -578,15 +613,23 @@ lv_read (void)
                     &stats, sizeof stats) != 0)
             continue;
 
+	char *new_path=NULL;
+	if( 0==strcasecmp("true", blockdevice_format_basename) && source==blockdevice_format){ //valid only if we use source (full path to the device)
+		new_path=strdup(basename(block_devices[i].path));
+	}else{
```

Comment:
Please mimic the style of the rest of the file, e.g. spaces around equal signs, spaces in front of opening parenthesis, curlies on their own line.